### ISSUE TITLE
Feature Request: Amazon EKS on AWS Fargate

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#11110]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Today, AWS has announced the release of Amazon EKS for AWS Fargate. HashiCorp is proud to announce launch day support for this feature in Terraform. Amazon EKS and Fargate make it straightforward to run Kubernetes-based applications on AWS by removing the need to provision and manage infrastructure for pods.

With AWS Fargate, customers don’t need to be experts in Kubernetes operations to run a cost-optimized and highly-available cluster. Fargate eliminates the need for customers to create or manage EC2 instances for their Amazon EKS clusters.

Customers no longer have to worry about patching, scaling, or securing a cluster of EC2 instances to run Kubernetes applications in the cloud. Using Fargate, customers define and pay for resources at the pod-level. This makes it easy to right-size resource utilization for each application and allow customers to clearly see the cost of each pod.

Available in version 2.41.0 of the Terraform AWS Provider, a new `aws_eks_fargate_profile` resource can manage this new functionality. The resource documentation can be found [on the Terraform website](https://www.terraform.io/docs/providers/aws/r/eks_fargate_profile.html).

### New Resource(s)

- `aws_eks_fargate_profile`

### Potential Terraform Configuration

```hcl
resource "aws_eks_fargate_profile" {
  cluster_name           = aws_eks_cluster.example.name           
  name                   = "example_fargate_profile"                   
  pod_execution_role_arn = aws_iam_role.example.arn 
  subnet_ids             = aws_subnet.example[*].id          
  tags = {
    TagKey1 = “TagValue1”
  }                   

  selectors { 
    labels = {
      LabelKey1 = “LabelValue1”
    }    
    namespace = "example-namespace" 
  }
}
```

### References

- Announcement Whats New Post: TBD
- [Announcement Blog](https://aws.amazon.com/blogs/aws/amazon-eks-on-aws-fargate-now-generally-available/)
- EKS Documentation: TBD
- AWS Go SDK: TBD

**Comment by bflad** [comment]
Please note we are waiting for the AWS Go SDK release that contains this functionality, then we can release it with the Terraform AWS Provider.

**Comment by unacceptable** [comment]
This looks to be a duplicate of #11109.

**Comment by MrGossett** [comment]
AWS Go SDK Release v1.25.46 was merged about 2 hours ago. It adds [Fargate for EKS](https://github.com/aws/aws-sdk-go/pull/2986/files?file-filters%5B%5D=.go#diff-1995b57df8fc20a561f06455134e6c0e).

**Comment by bflad** [comment]
A new `aws_eks_fargate_profile` resource has been merged and will release with version 2.41.0 of the Terraform AWS Provider, later today or tomorrow.
